pkg/resource/stack/deployment.go

```diff
@@ -165,7 +166,9 @@ func SerializeDeployment(snap *deploy.Snapshot, sm secrets.Manager, showSecrets
 // from it. DeserializeDeployment will return an error if the untyped deployment's version is
 // not within the range `DeploymentSchemaVersionCurrent` and `DeploymentSchemaVersionOldestSupported`.
 func DeserializeUntypedDeployment(
```

Comment:
Wondering if it's fair game to change the signature here or need to add a helper `*WithContext` to not break the API.